powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/task/CleaningTask.java

```diff
@@ -69,4 +72,20 @@ public void expireActivations() {
         activationServiceBehavior.expireActivations();
     }
 
+    @Scheduled(fixedRateString = "${powerauth.service.scheduled.job.retryFailedCallbackUrlEvent:3000}")
+    @SchedulerLock(name = "retryFailedCallbackUrlEvent")
+    public void callbackEvents() {
+        LockAssert.assertLocked();
+        logger.debug("dispatchFailedCallbackUrlEvent");
+        callbackUrlEventDispatcher.dispatchFailedCallbackUrlEvent();
+    }
+
+    @Scheduled(cron = "${powerauth.service.scheduled.job.callbackUrlEventsCleanupCron:0 0 0 */3 * *}")
+    @SchedulerLock(name = "callbackUrlEventsCleanup")
+    public void cleanEvents() {
+        LockAssert.assertLocked();
+        logger.debug("deleteCallbackUrlEvents");
+        callbackUrlEventDispatcher.deleteCallbackUrlEventsAfterRetentionPeriod();
```

Comment:
🤔 When I see calling the _dispatcher_ from _scheduled_ task, I suggest splitting the listener and the logic into two classes (dispatcher and service).

Reply:
for now I renamed dispatcher to service, as a separated service class would just contain two methods with a single call to dispatcher

Reply:
The new service is fine. I do not insist, but still think that `TransactionalEventListener` could be in a separate class (some _listener_?). @romanstrobl Any opinion?